src/gtfs_rt_utils.rs

```diff
@@ -120,6 +120,33 @@ fn create_stop_time_updates(
     Ok(res)
 }
 
+fn get_date(trip: &transit_realtime::TripDescriptor) -> chrono::ParseResult<chrono::NaiveDate> {
+    let default_date = chrono::Utc::today().format("%Y%m%d").to_string();
+    let date_str = trip.start_date.as_ref().unwrap_or(&default_date);
+    return chrono::NaiveDate::parse_from_str(&date_str, "%Y%m%d");
+}
```

Comment:
You compute default_date even if not needed.

return at the end of a function is not idiomatic.

```rust
 fn get_date(trip: &transit_realtime::TripDescriptor) -> chrono::ParseResult<chrono::NaiveDate> {
    trip.start_date
        .as_ref()
        .map(|s| chrono::NaiveDate::parse_from_str(s))
        .unwrap_or_else(|| chrono::Utc::today().naive_utc())
}
```

This UTC is smelly, I suspect you want the day of the GTFS timezone.

Reply:
I have problem with this one.

In the code you're proposing, you're returning today's date if there's a parsing error, this is not my intent, I want to return today's date if there's not date.

If  I don't want to compute today's date every time I came up with something like this : 
```rust
let date_str = trip.start_date.as_ref()
        .unwrap_or_else(|| &chrono::Utc::today().format("%Y%m%d").to_string());

    chrono::NaiveDate::parse_from_str(&date_str, "%Y%m%d")
```

Which is obviously not compiling because I create a temp value and return a ref on it.

Do you have something to suggest?

Reply:
I think you should do:

```rust

fn get_date(trip: &transit_realtime::TripDescriptor) -> Result<chrono::NaiveDate, failure::Error> {
    trip.start_date.as_ref().map_or_else(
        || Ok(chrono::Utc::today().naive_utc()),
        |s| {
            chrono::NaiveDate::parse_from_str(s, "%Y%m%d")
                .map_err(|e| format_err!("Impossible to parse date: {}", e))
        },
    )
}
```

and then use it below:

`let date = get_date(trip)?`

Reply:
I have updated the code in my comment, @antoine-de 's solution should also work.